include/deal.II/base/index_set.h

```diff
@@ -1218,6 +1232,44 @@ IndexSet::IndexSet (const size_type size)
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+
+inline
+IndexSet::IndexSet (IndexSet &&is)
+  :
+  ranges (std::move(is.ranges)),
+  is_compressed (is.is_compressed),
+  index_space_size (is.index_space_size),
+  largest_range (is.largest_range)
+{
+  is.clear ();
+  is.index_space_size = 0;
+  is.largest_range = numbers::invalid_unsigned_int;
```

Comment:
I don't recall what `clear()` does, but this is the thing that could potentially go wrong because after the move of the `ranges` variable, the moved-from object is left in an inconsistent state (the `largest_range` variable, for example, does not point to a valid `ranges` index any more). Calling functions whose internal implementation we don't know here is a risk in such situations. My preference would be to just work on member variables and not call member functions.